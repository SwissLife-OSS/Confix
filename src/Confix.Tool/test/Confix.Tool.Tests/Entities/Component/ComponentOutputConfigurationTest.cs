using System.Text.Json.Nodes;
using Confix.Tool.Abstractions;

namespace Confix.Entities.Component.Configuration;

public class ComponentOutputConfigurationTest : ParserTestBase
{
    [Fact]
    public void Parse_Should_ReturnValidComponentOutputConfiguration_When_ValidJsonNodeProvided()
    {
        ExpectValid(
            """
            {
                "type": "graphql",
                "additional": "property"
            }
        """);
    }

    [Fact]
    public void Parse_Should_BeValid_WhenEmpty()
    {
        ExpectValid(""" { } """);
    }

    [Fact]
    public void Parse_ShouldBeValid_When_AdditionalIsMissing()
    {
        ExpectValid(
            """
            {
                "type": "graphql"
            }
         """);
    }

    public override object Parse(JsonNode json)
        => ComponentOutputConfiguration.Parse(json);
}
