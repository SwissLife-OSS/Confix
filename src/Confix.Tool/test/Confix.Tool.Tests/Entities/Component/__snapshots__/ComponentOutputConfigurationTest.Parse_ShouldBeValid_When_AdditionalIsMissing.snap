{
  "Type": "graphql",
  "Value": {
    "type": "graphql"
  }
}
